ui/src/main/java/com/hyperwallet/android/ui/transfermethod/SelectTransferMethodPresenter.java

```diff
@@ -69,16 +77,41 @@ public void onKeysLoaded(@Nullable final HyperwalletTransferMethodConfigurationK
                                 if (!mView.isActive()) {
                                     return;
                                 }
+
+                                Country country = TextUtils.isEmpty(countryCode) ? key.getCountry(user.getCountry())
+                                        : key.getCountry(countryCode);
+
+                                if (country == null) { // param and user country is null
+                                    country = key.getCountry(DEFAULT_COUNTRY_CODE);
+                                }
+
+                                String currencyCodeString = currencyCode;
+                                Set<Currency> currencies = key.getCurrencies(country.getCode());
+                                if (TextUtils.isEmpty(currencyCodeString) && currencies != null
+                                        && !currencies.isEmpty()) {
+                                    currencyCodeString = ((Currency) currencies.toArray()[0]).getCode();
```

Comment:
Could you clarify conversion to an array there, as far as I know Sets are unordered. We may expect unpredictable results there.